ant/src/org/jetbrains/kotlin/ant/KotlinCompilerAdapter.java

```diff
@@ -87,15 +87,24 @@ public boolean execute() throws BuildException {
 
         kotlinc.getAdditionalArguments().addAll(additionalArguments);
 
-        kotlinc.execute();
-        if (!Integer.valueOf(0).equals(kotlinc.getExitCode())) {
-            // Don't run javac if failOnError = false and there were errors on Kotlin sources
-            return false;
+        // Javac13#execute passes everything in compileList to javac, which doesn't recognize .kt files
+        File[] compileListForJavac = filterOutKotlinSources(compileList);
+
+        boolean hasKotlinFilesInSources = compileListForJavac.length < compileList.length;
+
+        if (hasKotlinFilesInSources) {
+            kotlinc.execute();
+            if (!Integer.valueOf(0).equals(kotlinc.getExitCode())) {
+                // Don't run javac if failOnError = false and there were errors on Kotlin sources
+                return false;
+            }
+        }
+        else {
+            kotlinc.fillArguments();
         }
 
         javac.log("Running javac...");
 
-        // Javac13#execute passes everything in compileList to javac, which doesn't recognize .kt files
         compileList = filterOutKotlinSources(compileList);
```

Comment:
That was a mistake. Of course it should be `compileList = compileListForJavac;`.
https://github.com/JetBrains/kotlin/commit/e7958c1b10b9efe28f35e0fc1b058b682f150efb